pkg/networkmanager/interfaces.js

```diff
@@ -107,8 +107,16 @@ function select_btn_selected(btn) {
 }
 
 function connection_settings(c) {
-    var settings = (c || { }).Settings || { };
-    return settings.connection || { };
+    if (c && c.Settings && c.Settings.connection) {
+        return c.Settings.connection;
+    } else {
+        // It is a programming error if we ever access a Connection
+        // object that doesn't have it's settings yet, and we expect
+        // each Connection object to have "connection" settings.
+        console.warn("Incomplete 'Connection' object accessed", c);
+        try { throw new Error(); } catch (e) { console.log(e.stack); }
```

Comment:
OOI, is `console.trace()` still not portable enough? https://developer.mozilla.org/de/docs/Web/API/Console/trace looks quite fine to me?